app/web/features/auth/deletion/DeleteAccount.tsx

```diff
@@ -0,0 +1,121 @@
+import { Typography, useMediaQuery, useTheme } from "@material-ui/core";
+import Alert from "components/Alert";
+import Button from "components/Button";
+import TextField from "components/TextField";
+import { Empty } from "google-protobuf/google/protobuf/empty_pb";
+import { RpcError } from "grpc-web";
+import { Trans, useTranslation } from "i18n";
+import { AUTH, GLOBAL } from "i18n/namespaces";
+import { useForm } from "react-hook-form";
+import { useMutation } from "react-query";
+import { service } from "service";
+import { lowercaseAndTrimField } from "utils/validation";
+
+import useChangeDetailsFormStyles from "../useChangeDetailsFormStyles";
+
+interface DeleteAccountForm {
+  confirmUsername: string;
+  reason?: string;
+}
+
+interface DeleteAccountProps {
+  username: string;
+  className?: string;
+}
+
+export default function DeleteAccount({
+  className,
+  username,
+}: DeleteAccountProps) {
+  const { t } = useTranslation([AUTH, GLOBAL]);
+  const formClasses = useChangeDetailsFormStyles();
+  const theme = useTheme();
+  const isMdOrWider = useMediaQuery(theme.breakpoints.up("md"));
+
+  const {
+    handleSubmit,
+    register,
+    reset: resetForm,
+  } = useForm<DeleteAccountForm>();
+  const onSubmit = handleSubmit((data) => {
+    deleteAccount(data);
+  });
+
+  const {
+    error: deleteAccountError,
+    isLoading: isDeleteAccountLoading,
+    isSuccess: isDeleteAccountSuccess,
+    mutate: deleteAccount,
+  } = useMutation<Empty, RpcError, DeleteAccountForm>(
+    ({ confirmUsername, reason }) => {
+      const confirm =
+        lowercaseAndTrimField(confirmUsername) ===
+        lowercaseAndTrimField(username);
+      if (!confirm) {
+        throw Error(t("auth:delete_account.request.username_mismatch"));
+      }
+      return service.account.deleteAccount(confirm, reason);
+    },
+    {
+      onSuccess: () => {
+        resetForm();
```

Comment:
redirect to `logoutRoute`? Or what happens next time they try to make an api call?

Reply:
Yes, good idea. Though that's in `ConfirmDeleteAccount.tsx`